### PR TITLE
Fix feedrate after toolchange

### PR DESCRIPTION
**Pull request by Blue-Marlin**
The fix is simple. 
Most of the changes are because of changed indentation,
bacause of leaving early a for wrong tool number.
https://github.com/MarlinFirmware/Marlin/pull/3620/files?w=1

Fix for #3610

### CODE REVIEW

**Comment by Alex9779** [comment]
Thanks, when I got time, maybe later today...

**Comment by Alex9779** [comment]
Ok I tested. This PR fixes the issue I reported in #3610...

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -6304,131 +6304,136 @@ inline void gcode_T(uint8_t tmp_extruder) {
...
+    #else
+      feedrate = min(max_feedrate[X_AXIS], max_feedrate[Y_AXIS]);
+    #endif
+  }
 
-      #if EXTRUDERS > 1
-        make_move = true;
-      #endif
+  #if EXTRUDERS > 1
+    target_extruder = tmp_extruder;
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: It looks like this line doesn't do anything, so it can probably be dropped.

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -6304,131 +6304,136 @@ inline void gcode_T(uint8_t tmp_extruder) {
...
+        set_axis_is_at_home(X_AXIS);
 
-          if (dual_x_carriage_mode == DXC_FULL_CONTROL_MODE) {
+        if (dual_x_carriage_mode == DXC_FULL_CONTROL_MODE) {
+          current_position[X_AXIS] = inactive_extruder_x_pos;
+          inactive_extruder_x_pos = destination[X_AXIS];
+        }
+        else if (dual_x_carriage_mode == DXC_DUPLICATION_MODE) {
+          active_extruder_parked = (active_extruder == 0); // this triggers the second extruder to move into the duplication position
+          if (active_extruder == 0 || active_extruder_parked)
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: Because `active_extruder_parked = (active_extruder == 0)`…

``` cpp
if (active_extruder_parked) ...
```

**Comment by Blue-Marlin** [comment]
@thinkyhead 
Updated.

**Comment by thinkyhead** [comment]
This is nice and comprehensive. I have my fingers crossed that it works well.
We don't get to test DUAL_X nearly enough!

**Comment by thinkyhead** [comment]
> Most of the changes are because of changed indentation

I have found it helpful in such cases to do the code changes first, but leave indentation the same, commit that first. And then commit the indentation changes (elimination of `else`, adding `return`, etc.) afterward. Then at least the first commit makes a more readable diff.

**Pull request merged**